csrc/scheduler/mma_utils.cpp

```diff
@@ -1029,63 +1023,62 @@ void WarpMmaSwizzler::scheduleOperandRead(
   tv->setAllocationDomain(tv->getLoopDomain(), true);
 }
 
-void WarpMmaSwizzler::scheduleMmaWarpOutput(TensorView* tv) {
+AbstractTensor MmaSwizzler::scheduleMmaOutputAllocation(AbstractTensor t) {
   // This function works for all mma ops, regardless of the architecture. The
   // Hopper one is the most general one. For earlier architectures, we will have
   // some dimensions with size 1 after split, this is fine.
   // Memory format for hopper mma:
   // https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#wgmma-64n16-d
 
   // Assume last 2 dims, for example [M64, N24] or [M64, N24, R]
-  NVF_ERROR(tv->nDims() >= 2);
-  bool is_mma_output = tv->definition()->isA<MmaOp>();
+  NVF_ERROR(t.size() >= 2);
+  bool has_reduction = t[-1]->isReduction();
```

Comment:
Should this be an assert now? My understanding is that we no longer apply this to anything but `mma_result`.

Reply:
No, today, we still apply this function to the consumer of `mma_result` in some unit tests. I believe these tests do so because this is just a convenient utility to manually schedule some tensor for prototyping purposes. For example, if I have a fusion:

```C++
T4_l = mma(...);
T5_g = set(T4_l);
```

Then I can just call this function on `T5_g` to schedule it like `T4_l`. For this PR, let's preserve this behavior. But in the future, I am not a fan of supporting both variants. So I don't mind create a separate PR to clean this up.

Reply:
For non-reduction tensor, I believe we can just push a placeholder as the reduction dimension: https://github.com/NVIDIA/Fuser/pull/2725